pkg/adaptor/cloud/azure/provider.go

```diff
@@ -291,6 +315,23 @@ func (p *azureProvider) CreateInstance(ctx context.Context, podName, sandboxID s
 		return nil, err
 	}
 
+	// Add tags to the instance
+	tags := map[string]*string{
+		"Name": to.Ptr(instanceName),
+	}
+
+	// Add custom tags from serviceConfig.Tags to the instance
+	for k, v := range p.serviceConfig.Tags {
+		tags[k] = to.Ptr(v)
+	}
+
+	// Call the addTags method to add tags to the instance
+	err = p.addTags(ctx, instanceName, tags)
```

Comment:
Why are we adding the tags after the VM is created, as in why not do it while the VM create request is sent?

https://github.com/confidential-containers/cloud-api-adaptor/blob/b03177aa1ddeb121cd5e983d88c4d6b1e8a9a135/pkg/adaptor/cloud/azure/provider.go#L114-L129

This approach increases the number of calls we make to the API.

Reply:
Fair point. I'll see if I can add it during the VM creation call itself.